tests/integration/targets/aci_aep_to_epg/tasks/main.yml

```diff
@@ -0,0 +1,213 @@
+    aci_info: &aci_info
+      host: '{{ aci_hostname }}'
+      username: '{{ aci_username }}'
+      password: '{{ aci_password }}'
+      validate_certs: '{{ aci_validate_certs | default(false) }}'
+      use_ssl: '{{ aci_use_ssl | default(true) }}'
+      use_proxy: '{{ aci_use_proxy | default(true) }}'
+      output_level: '{{ aci_output_level | default("info") }}'
+
+# CLEAN ENVIRONMENT
+- name: Remove test aep before we kickoff
+  cisco.aci.aci_aep: &aep_absent
+    <<: *aci_info
+    aep: epg2aep
+    state: absent
+
+# SETUP TEST AEP AND EPGS
+- name: Create AEP
+  cisco.aci.aci_aep: &aep_present
+    <<: *aci_info
+    aep: epg2aep
+    state: present
+
+- name: Create Tenant
+  cisco.aci.aci_tenant: &tenant_present
+    <<: *aci_info
+    tenant: epg2aep
+    state: present
+
+- name: Configure VRF
+  cisco.aci.aci_vrf:
+    <<: *tenant_present
+    vrf: epg2aep
+
+- name: Configure BD
+  cisco.aci.aci_bd:
+    <<: *tenant_present
+    vrf: epg2aep
+    bd: epg2aep
+
+- name: Configure Ap
+  cisco.aci.aci_ap:
+    <<: *tenant_present
+    ap: ap
+
+- name: Configure EPG
+  cisco.aci.aci_epg:
+    <<: *tenant_present
+    ap: ap
+    epg: '{{ item }}'
+    bd: epg2aep
+  loop:
+    - epg1
+    - epg2
+
+# ADD ASSOCIATION
+- name: Add EPG to AEP association (check_mode)
+  cisco.aci.aci_aep_to_epg: &assoc_present
+    <<: *aep_present
+    tenant: epg2aep
+    ap: ap
+    epg: epg1
+    encap: 1984
+    interface_mode: trunk
+  check_mode: yes
+  register: cm_add_assoc
+
+- name: Add EPG to AEP association (normal mode)
+  cisco.aci.aci_aep_to_epg:
+    <<: *assoc_present
+  register: nm_add_assoc
+
+- name: Verify add_assoc
+  assert:
+    that:
+    - cm_add_assoc is changed
+    - nm_add_assoc is changed
+    - cm_add_assoc.previous == nm_add_assoc.previous == []
+    - cm_add_assoc.sent.infraGeneric.children[0].infraRsFuncToEpg.attributes.encap == nm_add_assoc.sent.infraGeneric.children[0].infraRsFuncToEpg.attributes.encap == 'vlan-1984'
+    - cm_add_assoc.sent.infraGeneric.children[0].infraRsFuncToEpg.attributes.mode == nm_add_assoc.sent.infraGeneric.children[0].infraRsFuncToEpg.attributes.mode == 'regular'
+    - cm_add_assoc.sent.infraGeneric.children[0].infraRsFuncToEpg.attributes.tDn == nm_add_assoc.sent.infraGeneric.children[0].infraRsFuncToEpg.attributes.tDn == 'uni/tn-epg2aep/ap-ap/epg-epg1'
+
+- name: Add EPG to AEP association again, check if idempotency works (check_mode)
+  cisco.aci.aci_aep_to_epg:
+    <<: *assoc_present
+  check_mode: yes
+  register: cm_add_assoc_again
+
+- name: Add EPG to AEP association again, check if idempotency works (normal_mode)
+  cisco.aci.aci_aep_to_epg:
+    <<: *assoc_present
+  register: nm_add_assoc_again
+
+- name: Verify add_assoc_again
+  assert:
+    that:
+    - cm_add_assoc_again is not changed
+    - nm_add_assoc_again is not changed
+
+# UPDATE ASSOCIATION
+- name: Change encap (check_mode)
+  cisco.aci.aci_aep_to_epg: &assoc_update
+    <<: *assoc_present
+    encap: 1985
```

Comment:
Can you add primary_encap attribute here?
L290-291 in Python file are not currently tested by adding the primary_encap attribute here and checking value in the "Verify update_assoc" task that will cover it fully.